src/Microsoft.ML.Transforms/MissingValueReplacingUtils.cs

```diff
@@ -340,6 +362,51 @@ protected long GetValuesProcessed(int slot)
             protected abstract void ProcessValueMax(in TItem val, int slot);
         }
 
+        /// <summary>
+        /// A mutable struct for keeping the appropriate statistics for mode calculations, whose scope is restricted
+        /// and only exists as an instance in a field or an array, utilizing the mutation of the struct correctly.
+        /// </summary>
+        private class ModeStat<TType>
+        {
+            // Delegate used to check if the value is valid. We use a delegate so that this class can support modes of all types.
+            public delegate bool IsValid(in TType val);
+
+            private TType _modeSoFar;
+            private int _maxCount;
+            private Dictionary<TType, int> _valueCounts;
+            private IsValid _validityCheck;
+            public ModeStat(IsValid valid)
+            {
+                _modeSoFar = default;
+                _maxCount = 0;
+                _valueCounts = new Dictionary<TType, int>();
+                _validityCheck = valid;
+            }
+
+            public void Update(TType val)
+            {
+                // We don't include non finite values in the mode, so if its not finite then just return.
+                if (!_validityCheck(val))
+                    return;
+
+                if (_valueCounts.TryGetValue(val, out int count))
+                    count++;
+
+                _valueCounts[val] = count;
+
```

Comment:
When TryGetValue returns false, count is initialized to 0 and _valueCounts[val] will be set to 1. But in the previous iteration, the default value it was set to was zero. 

The CI tests passed in both cases.  What should be the right behavior? Is this code path being exercised?

Reply:
Both of the cases did the same thing. This just does it with less lines of code.
Before it was this
```
int count = 1;
if (_valueCounts.TryGetValue(val, out int curCount))
    count = curCount + 1;

_valueCounts[val] = count;
```
It was still being initialized to 1, just using an extra variable.

Reply:
Fixed based on our offline discussion. Added the else clause so count is always initialized to 1 and a comment.